src/contract.rs

```diff
@@ -56,6 +57,47 @@ impl Contract for ApplicationContract {
 }
 
 impl ApplicationContract {
+    /// Handles an [`Operation::UpdateNodes`] by adding the `nodes_to_add` and removing the
+    /// `nodes_to_remove`.
+    fn update_nodes(&mut self, nodes_to_add: Vec<PublicKey>, nodes_to_remove: Vec<PublicKey>) {
```

Comment:
Wouldn't it be simpler to pass the set of currently-active nodes?

Reply:
We can do so at this point, in the future, once the network scales, we might leave as a permissionless endpoint where the nodes can subscribe to Atoma Linera micro-chain ?